src/components/Card.js

```diff
@@ -30,32 +41,50 @@ export default class Card {
   }
 
   _setEventListeners() {
+    this._likeButton = this._element.querySelector(".card__like-button");
     this._element
       .querySelector(".card__image")
       .addEventListener("click", () => {
         this._handleCardImageClick(this._name, this._link);
       });
 
-    this._element
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleCardLike();
-      });
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeCard(this);
+    });
 
     this._element
       .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleCardDelete();
+        this._handleDeleteCard(this, this._id);
       });
   }
 
-  _handleCardLike() {
-    this._element
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
-  }
-
   _handleCardDelete() {
     this._element.remove();
   }
+
+  getId() {
+    return this._id;
+  }
+
+  isLiked() {
+    return this._isLiked;
+  }
+
+  renderLike() {
+    if (this._isLiked) {
+      this._element
+        .querySelector(".card__like-button")
```

Comment:
You've already found `this._likeButton` in line 44. Please, use it everywhere

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.